src/app/views/query-runner/query-input/auto-complete/auto-complete.util.ts

```diff
@@ -58,12 +58,16 @@ function getErrorMessage(queryUrl: string) {
 }
 
 function getValidationError(queryUrl: string): string | null {
-  const validator = new ValidatedUrl();
-  const validation = validator.validate(queryUrl);
-  if (!validation.success) {
-    return queryUrl.substring(validation.matched, validation.maxMatched);
+  try {
+    const validator = new ValidatedUrl();
+    const validation = validator.validate(queryUrl);
+    if (!validation.success) {
+      return queryUrl.substring(validation.matched, validation.maxMatched);
+    }
+    throw new Error('Validation unsuccessful');
```

Comment:
It seems strange that you throw an error when validation is successful then immediately catch it. Am I misunderstanding something here?

Reply:
The PR description made it seem like the BNF library raised the exception, but here it looks like the exception is raised manually...why not just return null instead of throwing then catching and returning null? The code is confusing me.

Reply:
Yes, I see why this can be confusing. I've changed it to return null